# Protein Secondary Structure Prediction with Neural Networks
*Individual Assignment - Bioinformatics Course*

## Overview
This repository contains the code, data and report for a project conducted as part of a master's bioinformatics course. The project aimed to enhance protein secondary structure prediction using neural networks. The code is implemented in Python and executed in Jupyter Notebook.

## Project Description
The project utilized neural networks to predict protein secondary structures from amino acid sequences. It incorporated multiple sequence alignment data and employed a sliding-window approach to improve prediction accuracy. Various techniques, including dropout and ensemble modeling, were explored to optimize model performance and mitigate overfitting.

## Repository Structure
- Jupyter Notebook: Contains the Python code for the project.
- Data: Includes datasets used for training, cross-validation, and testing the models. The data were sourced from Katarina Elez's GitHub repository at https://github.com/katarinaelez/protein-ss-pred.
- Results: Contains visualizations and metrics evaluating model performance.
- Documentation: Additional documentation and resources related to the project.

## Usage
To run the code, ensure you have Python installed along with necessary libraries like NumPy, Pandas, Matplotlib, Seaborn, TensorFlow, Keras, and scikit-learn. Clone the repository and execute the Jupyter Notebook files in the 'Notebooks' directory.
